src/LineUI/LineUnit.tsx

```diff
@@ -146,6 +152,113 @@ const LineUnit : React.FC<ILineUnitProps> = (props) => {
     x: (x2 + x1) / 2,
     y: (y2 + y1) / 2
   };
+
+  const directionMarkCoordinate = () => {
+    const angle = Math.atan2((y2 - y1), (x2 - x1));
+    const angleMode = (Math.PI / 2) - angle;
+    const x = midpoint.x + Math.sin(angleMode) -5;
+    const y = midpoint.y + Math.cos(angleMode);
+    let rotate = (180 / Math.PI) * Math.atan2(y2 - y1, x2 - x1);
+
+    let labelX = 0;
+    let labelY = isUpDirection ? 30 : 20;
+    let labelRotate = 0;
+    let rotatefactor = isUpDirection ? -3 : -8;
+    
+    const lblMargin = label ? (label.length * 16)/2 : 0;
+    isUpDirection ? labelX = -(lblMargin/2) : labelX = (lblMargin/2);
+    if(rotate >=0){
+      if(rotate >30 && rotate <90){
+        if(isUpDirection){
+          labelY += 10;
+        }
+        labelRotate = -(rotate/3);
+      } else if(rotate >=90 && rotate <120){
+        if(isUpDirection){
+          labelY += 20;
+        }
+        labelX = 0;
+        labelRotate = -70;
+      } else if(rotate >=120 && rotate <=180){
+        if(isUpDirection){
+          labelX = (lblMargin/2);
+          labelY = 20;
+          rotatefactor = -8;
+        } else {
+          labelX = -(lblMargin/2);
+          labelY = 30;
+          rotatefactor = -3;
+        }
+        labelRotate = 180;
+      }
+    } else {
+      if(isUpDirection){
+        labelX = (lblMargin/2);
+        labelY = 20;
+        rotatefactor = -8;
+      } else {
+        labelX = -(lblMargin/2);
+        labelY = 30;
+        rotatefactor = -3;
+      }
+      labelRotate = 180;
+      
+      if(rotate < -90 && rotate >= -150){
+        labelRotate = 150;
+        if(isUpDirection){
+          labelX = (lblMargin/2);
+          labelY = 20;
+        } else {
+          labelX = -(lblMargin/2);
+          labelY = 35;
+        }
+      } else if(rotate < -45 && rotate >= -90){
+        labelRotate = 45;
+        if(isUpDirection){
+          labelX = -(lblMargin/2);
+          labelY = 25;
+          rotatefactor = -8;
+        } else {
+          labelX = (lblMargin/2);
+          labelY = 35;
+          rotatefactor = -8;
+        }
+      } else if(rotate < 0 && rotate >= -45){
+        labelRotate = 0;
+        
+        if(isUpDirection){
+          labelX = -(lblMargin/2);
+          labelY = 30;
+          rotatefactor = -3;
+        } else {
+          labelX = (lblMargin/2);
+          labelY = 20;
+          rotatefactor = -8;
+        }
+      }
+    }
+    rotate = isUpDirection ? rotate : rotate + 180;
+    labelRotate = isUpDirection ? labelRotate : labelRotate + 180;
+    return {x, y, rotate, labelX, labelY, labelRotate, rotatefactor};
+  };
```

Comment:
Remove any code that is manipulating the label position. Let's use the current behaviour for labels.

Reply:
Code from line 163 to 241 is removed.